.github/workflows/schedule_maintenance.yml

```diff
@@ -0,0 +1,64 @@
+name: schedule_maintenance
+
+on:
+  schedule:
+    - cron: '0 9 1 * *'  # Every month on the first day at 09:00 UTC
+
+permissions:
+  contents: write
+  issues: write
+
+jobs:
+  create-issue:
+    name: Create an issue
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+          cache: "pip"
+
+      - name: Install dependencies
+        run: |
+          pip install ruamel.yaml
+        # If there is any output (non-empty), set should_create_issue to true.
+        # Otherwise, set it to false.
+        # Then export the DATASETS variable to the GitHub Actions environment
+        # so it can be accessed later, when creating the issue.
+      - name: Generate datasets list
+        id: generate
+        run: |
+          DATASETS=$(python contrib/schedule_maintenance.py)
+          # the heredoc (<<EOF) syntax is to set a multiline value as env var
+          echo "DATASETS<<EOF" >> $GITHUB_ENV
+          echo "$DATASETS" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Commit and push changes
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
```

Comment:
I think the numeric ID is supposed to be part of the email address https://github.com/actions/checkout?tab=readme-ov-file#Push-a-commit-using-the-built-in-token